### ISSUE TITLE
How to improve and modify the ZH documentation

### ISSUE THREAD

**Issue by lushikai** [ultralytics/ultralytics#13652]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello @glenn-jocher I had some problems preparing to revise and improve the Chinese documentation, and I didn't know where to start changing the Chinese documentation.
1. Command may have a problem?
```
mkdocs build -f docs/mkdocs.yml
```
But the 'mkdocs.yml' is under 'ultralytics/mkdocs.yml ' instead of 'ultralytics/docs/mkdocs.yml'. I successfully build the 'site' folder using command 'mkdocs build -f mkdocs.yml'.
2. How to use these script in the terminal
```
for file in docs/mkdocs_*.yml; do
  echo "Building MkDocs site with $file"
  mkdocs build -f "$file"
done
```
3. Difference between command 'mkdocs serve' and 'python -m http.server'. How to use in correct scenario.
4. How to deal with warning
![image](https://github.com/ultralytics/ultralytics/assets/121033683/a2cd6449-53f9-4b95-8177-4d4aeca22d62)
5. When I run the service locally, the website starts correctly on port number 8000, but I switch the Chinese document and still jumps to the normal website. And I can't find an entry on how to improve and modify the Chinese documentation.
![image](https://github.com/ultralytics/ultralytics/assets/121033683/3a058db8-6a79-4319-8d8d-11d0c7e0f41c)

![image](https://github.com/ultralytics/ultralytics/assets/121033683/6edee5af-ff71-4e07-aca1-c65624b25e7d)

**Comment by glenn-jocher** [comment]
Hello @lushikai,

Thank you for your interest in improving the Chinese documentation for YOLOv8! Let's address your questions step-by-step:

1. **Command Issue**:
   - You are correct that the `mkdocs.yml` file is located at `ultralytics/mkdocs.yml`. The correct command to build the site is:
     ```bash
     mkdocs build -f mkdocs.yml
     ```
   - This should generate the `site` folder successfully.

2. **Using Scripts in the Terminal**:
   - The script you mentioned is a loop to build multiple MkDocs sites if there are multiple configuration files. You can run it in your terminal as follows:
     ```bash
     for file in docs/mkdocs_*.yml; do
       echo "Building MkDocs site with $file"
       mkdocs build -f "$file"
     done
     ```
   - This will iterate over each `mkdocs_*.yml` file in the `docs` directory and build the site for each configuration.

3. **Difference Between `mkdocs serve` and `python -m http.server`**:
   - `mkdocs serve`: This command serves your MkDocs site locally and automatically rebuilds it when changes are detected. It's ideal for development.
     ```bash
     mkdocs serve
     ```
   - `python -m http.server`: This command serves static files from a directory. It's useful for serving already built sites but does not rebuild on changes.
     ```bash
     python -m http.server
     ```
   - Use `mkdocs serve` during development to see changes in real-time, and `python -m http.server` for serving static files.

4. **Dealing with Warnings**:
   - The warning in your screenshot indicates a missing `site_url` in your `mkdocs.yml`. You can add it to your configuration file:
     ```yaml
     site_url: "http://localhost:8000"
     ```
   - This should resolve the warning.

5. **Switching to Chinese Documentation**:
   - Ensure that the Chinese documentation is correctly linked and accessible. Check the `mkdocs.yml` for proper configuration of language-specific documentation.
   - If the site still redirects to the default language, verify the links and navigation settings in your `mkdocs.yml` to ensure they point to the correct Chinese documentation files.

If you encounter any further issues or need more detailed guidance, feel free to ask. Your contributions to the documentation are greatly appreciated by the entire YOLO community! 😊

**Comment by lushikai** [comment]
@glenn-jocher I used the mkdocs tool for the first time. But the Ultralytics source code library only has the markdown of EN, and then uses the mkdocs tool to automatically translate it? I can't fix the reading difficulties caused by the translation error of the Chinese document. How do I proceed?

**Comment by glenn-jocher** [comment]
Hello @lushikai,

Thank you for reaching out! It's great to hear that you're diving into using MkDocs for documentation.

To address your concerns:

1. **Translation Process**:
   - MkDocs itself does not automatically translate documentation. If you need Chinese documentation, it must be manually translated and maintained alongside the English version.
   - You can create a separate directory for Chinese markdown files (e.g., `docs/zh`) and configure MkDocs to include this in the site navigation.

2. **Handling Translation Errors**:
   - Manual translation ensures accuracy and context-specific clarity. Automated translation tools can introduce errors that might confuse readers.
   - Consider using a professional translation service or collaborating with native speakers to ensure high-quality translations.

3. **Steps to Add Chinese Documentation**:
   - Create a directory for Chinese documentation:
     ```bash
     mkdir -p docs/zh
     ```
   - Translate your markdown files and place them in the `docs/zh` directory.
   - Update your `mkdocs.yml` to include the Chinese documentation in the navigation:
     ```yaml
     nav:
       - Home: index.md
       - 中文:
           - 主页: zh/index.md
           - 其他页面: zh/other_page.md
     ```

4. **Building and Serving the Site**:
   - Use the following command to build and serve your site locally:
     ```bash
     mkdocs serve
     ```
   - This will allow you to preview the documentation and ensure everything is correctly linked.

If you encounter any specific issues or need further assistance, feel free to ask. Your efforts to improve the documentation are highly valued by the community! 😊

**Comment by lushikai** [comment]
@glenn-jocher Thanks! I will start trying to make Chinese documents under the 'docs/zh' folder.

**Comment by lushikai** [comment]
For the file naming norms, I still follow the English naming norms. I first try to deploy some of the translated files in some folders, If you think my job is okay, I will continue to maintain the Chinese documentation for ultralytics! 😄

**Comment by glenn-jocher** [comment]
Hello @lushikai,

Thank you for your initiative in translating and maintaining the Chinese documentation for Ultralytics! 🌟

Your approach of following English naming norms and deploying translated files in specific folders sounds excellent. This will help maintain consistency and make it easier for others to navigate the documentation.

Feel free to continue with your work, and if you encounter any issues or need further guidance, don't hesitate to reach out. Your contributions are greatly appreciated by the entire YOLO community! 😊

Keep up the great work!

**Comment by lushikai** [comment]
@glenn-jocher Can I temporarily copy all the English document under the docs/zh folder? Because there are so many markdown documents, I can only translate the English markdown into Chinese slowly, and I want to create a new 'mkdocs_zh.yml' file under docs/zh referring mkdocs.yml. This minimizes changes to the mkdocs.yml and allows me to focus more on solving reading difficulties caused by translation errors. Or can you provide a concrete solution? 😺 😃

**Comment by glenn-jocher** [comment]
Hello @lushikai,

Thank you for your proactive approach to translating the documentation! 😊

Yes, you can certainly copy all the English markdown documents into the `docs/zh` folder. This will allow you to translate them gradually. Here's a step-by-step approach:

1. **Copy English Documents**:
   ```bash
   cp -r docs/* docs/zh/
   ```

2. **Create a New `mkdocs_zh.yml`**:
   - You can create a new `mkdocs_zh.yml` file in the root directory, similar to `mkdocs.yml`, but pointing to the `docs/zh` folder:
     ```yaml
     site_name: Ultralytics YOLOv8 (ZH)
     docs_dir: docs/zh
     nav:
       - Home: index.md
       - Getting Started: getting_started.md
       # Add other sections as needed
     ```

3. **Build and Serve the Chinese Documentation**:
   - Use the following command to build and serve the Chinese documentation:
     ```bash
     mkdocs serve -f mkdocs_zh.yml
     ```

This approach minimizes changes to the existing `mkdocs.yml` and allows you to focus on translating the documents at your own pace.

Feel free to reach out if you need further assistance. Your contributions are highly valued by the community! 🌟

**Comment by lushikai** [comment]
I'm having an issue where the '.gitignore' file is working causing the replicated 'datasets' folder to be ignored.
![image](https://github.com/ultralytics/ultralytics/assets/121033683/3df4259b-583e-4f9d-8e1a-6a4995b483f0)

**Comment by glenn-jocher** [comment]
Hello @lushikai,

Thank you for reaching out! It looks like you're experiencing an issue with the `.gitignore` file causing your `datasets` folder to be ignored.

To help you better, could you please provide a minimum reproducible code example? This will allow us to understand the issue more clearly and offer a precise solution. You can find more details on how to create one here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

Additionally, please ensure you are using the latest versions of `torch` and `ultralytics`. You can update your packages using:
```bash
pip install --upgrade torch ultralytics
```

Once we have the necessary details, we'll be able to investigate further. Thank you for your cooperation! 😊

**Comment by lushikai** [comment]
@glenn-jocher Because I temporarily want to copy all the English documents into the 'docs/zh' folder, so I don't need to create a new Chinese markdown file, I only need to translate the English markdown into Chinese and then deploy it with 'mkdocs' tool, but because the copy operation is operated after '.gitignore', it also causes the '.gitignore' file to work, resulting in the 'datasets' folder under the copied 'docs/zh' being ignored. It will not be pushed to the GitHub repository properly.

**Comment by lushikai** [comment]
![image](https://github.com/ultralytics/ultralytics/assets/121033683/7463f00c-85a8-401e-bba4-722a754604fb)

**Comment by glenn-jocher** [comment]
Hello @lushikai,

Thank you for your detailed explanation! It sounds like you're on the right track with your approach. To address the issue with the `.gitignore` file causing the `datasets` folder to be ignored, you can use the following steps:

1. **Modify `.gitignore`**:
   - Temporarily modify your `.gitignore` to allow the `datasets` folder under `docs/zh` to be tracked. You can do this by adding an exception rule:
     ```plaintext
     # Ignore all datasets folders
     datasets/

     # But not the one under docs/zh
     !docs/zh/datasets/
     ```

2. **Add the `datasets` Folder**:
   - After modifying the `.gitignore`, you can add the `datasets` folder to your Git repository:
     ```bash
     git add -f docs/zh/datasets
     ```

3. **Commit the Changes**:
   - Commit the changes to your repository:
     ```bash
     git commit -m "Add datasets folder under docs/zh"
     ```

4. **Revert `.gitignore` (Optional)**:
   - If you want to revert the `.gitignore` changes after committing, you can do so, but ensure that the `datasets` folder under `docs/zh` remains tracked.

This approach will allow you to keep the `datasets` folder under `docs/zh` while maintaining the original `.gitignore` rules for other `datasets` folders.

Feel free to reach out if you have any further questions or need additional assistance. Your contributions to the Chinese documentation are greatly appreciated! 😊

**Comment by jsanhuo** [comment]
Hi @glenn-jocher 
I am a software engineer from China, and I would also like to participate in the optimization project for the Chinese documentation. Could you please let me know which branch I should submit the PR for document-related changes? Alternatively, is it possible to create a new branch for submitting the Chinese documentation, and then merge it into the main branch when the translation work reaches 30% or 50% completion?

**Comment by glenn-jocher** [comment]
Hello @jsanhuo,

Thank you for your interest in contributing to the Chinese documentation! 🌟

You can submit your PR for document-related changes to the `main` branch. If you prefer to work on a separate branch until the translation reaches a certain level of completion, you can create a new branch in your forked repository and submit a PR to `main` once you're ready.

Here's a suggested workflow:
1. **Fork the Repository**: Fork the Ultralytics YOLO repository to your GitHub account.
2. **Create a Branch**: Create a new branch in your forked repository for your translation work.
   ```bash
   git checkout -b zh-translation
   ```
3. **Make Your Changes**: Translate the documentation and commit your changes.
4. **Submit a PR**: When you're ready, submit a PR from your `zh-translation` branch to the `main` branch of the Ultralytics repository.

This approach allows you to work independently while keeping your changes organized. Once your translation work reaches a significant milestone, you can submit the PR for review.

We appreciate your contributions and look forward to your PR! If you have any further questions, feel free to ask. 😊

**Comment by jsanhuo** [comment]
Hi @glenn-jocher,
Okay, I will submit a pull request to the main branch, as mentioned by @lushikai . I will make a copy of the "en" folder and rename it. Additionally, I will duplicate the "mkdocs.yml" file and rename it as "mkdocs_zh.yml".
This way, subsequent modifications to the Chinese translation can be traced, and it will also facilitate the review process for my initial large-scale file submission. You can simply compare it with the "en" folder.

**Comment by glenn-jocher** [comment]
Hello @jsanhuo,

That sounds like a great plan! Creating a copy of the "en" folder and renaming it, along with duplicating the "mkdocs.yml" file to "mkdocs_zh.yml", will indeed help in tracking the modifications and facilitate the review process.

Feel free to submit your pull request to the `main` branch once you're ready. We appreciate your efforts in contributing to the Chinese documentation and look forward to reviewing your PR. If you have any further questions or need assistance, don't hesitate to ask. 😊

Thank you for your valuable contributions!

**Comment by jsanhuo** [comment]
Hi @glenn-jocher,
I have created a pull request https://github.com/ultralytics/ultralytics/pull/13874 for initializing the Chinese documentation. Please help review it. Once it's reviewed, I will use this English version as a basis for translation.

**Comment by glenn-jocher** [comment]
Hello @jsanhuo,

Thank you for your initiative and effort in creating the pull request for the Chinese documentation! 🌟

We will review your PR as soon as possible. Using the English version as a basis for translation is a great approach. Once the initial review is complete, you can proceed with the translation work.

If you have any questions or need further assistance during the process, feel free to reach out. We appreciate your contributions to the Ultralytics community!

**Comment by lushikai** [comment]
@glenn-jocher Hello! How much do you think the completion of Chinese documents is suitable for uploading PR, I feel that I need some encouragement, and it will be more rewarding and easier to complete step by step. It takes more time and patience to complete the translation at once. 😺

**Comment by lushikai** [comment]
For example, if I can finish translating a couple of markdown files in the 'docs/en/models' folder and start the service correctly, I'll try to pull a PR?

**Comment by glenn-jocher** [comment]
Hello @lushikai,

That sounds like a fantastic approach! Translating a couple of markdown files in the 'docs/en/models' folder and starting the service correctly is a great way to make incremental progress. This will also allow for quicker reviews and feedback, making the process more manageable and rewarding.

Feel free to submit a PR once you've translated those files. This step-by-step method will help maintain momentum and ensure that each part of the documentation is thoroughly reviewed and polished.

Keep up the great work! 😊 If you have any questions or need further assistance, don't hesitate to reach out.

**Comment by glenn-jocher** [comment]
Hello @lushikai,

Great to hear that you've completed translating two files! 🎉

You can go ahead and pull a PR to the `main` branch with your translated files. This incremental approach allows for quicker reviews and feedback, making the process more manageable and rewarding. Once the initial translations are reviewed and merged, you can continue translating additional files and submit further PRs as you progress.

Thank you for your valuable contributions to the Ultralytics community! If you have any questions or need further assistance, feel free to reach out. 😊

**Comment by lushikai** [comment]
@glenn-jocher I've pushed the Chinese documentation to my forked repository, but I don't know how to pull the PR to be seen by you. and My translated file changes the English under nav to Chinese in the `mkdocs_zh.yml` file, which indicates that the file has been translated.
![QQ_1720548828203](https://github.com/ultralytics/ultralytics/assets/121033683/cda011cd-7dcf-4aea-8e2e-63d370d0829f)

**Comment by lushikai** [comment]
https://github.com/ultralytics/ultralytics/pull/14306

**Comment by glenn-jocher** [comment]
Hello @lushikai,

Thank you for your efforts in translating the documentation! To create a pull request (PR) that we can review, follow these steps:

1. **Navigate to Your Forked Repository**: Go to your forked repository on GitHub.
2. **Switch to Your Branch**: Ensure you are on the branch where you made the changes.
3. **Create a Pull Request**:
   - Click on the "Pull requests" tab.
   - Click the "New pull request" button.
   - Ensure the base repository is set to `ultralytics/ultralytics` and the base branch is `main`.
   - Compare it with your fork and branch.
   - Add a title and description for your PR, explaining the changes you made.
   - Click "Create pull request".

Regarding the `mkdocs_zh.yml` file, your approach to indicate translated files by changing the navigation text to Chinese is a good idea. This helps reviewers quickly identify which parts have been translated.

Feel free to submit your PR once you're ready. We look forward to reviewing your contributions! 😊

If you have any further questions or need assistance, don't hesitate to ask. Thank you for your valuable contributions to the Ultralytics community!
